windows/src/developer/kmconvert/Keyman.Developer.System.KMConvertParameters.pas

```diff
@@ -0,0 +1,107 @@
+unit Keyman.Developer.System.KMConvertParameters;
+
+interface
+
+uses
+  UKeymanTargets;
+
+type
+  TKMConvertMode = (cmImportWindows, cmTemplate);
+
+  TKMConvertParameters = record
+    KLID: string;
+    Destination: string;
+    KeyboardID: string;
+    Prefix: string;
+    Name: string;
+    Copyright: string;
+    Version: string;
+    BCP47Tags: string;
+    Author: string;
+    Targets: TKeymanTargets;
+    Mode: TKMConvertMode;
+  private
+    function CheckParam(name, value: string): Boolean;
+  public
+    procedure WriteUsage;
+    function CheckParams: Boolean;
+  end;
+
+implementation
+
+uses
+  System.SysUtils;
+
+function TKMConvertParameters.CheckParams: Boolean;
+var
+  ModeString: string;
+  i: Integer;
+begin
+  Destination := '.';
+  Copyright := 'Copyright (C)';
+  Version := '1.0';
+
+  if ParamCount < 2 then
+    Exit(False);
+
+  ModeString := LowerCase(ParamStr(1));
+  if ModeString = 'import-windows' then
+    Mode := cmImportWindows
+  else if ModeString = 'template' then
+    Mode := cmTemplate
+  else
+    Exit(False);
+
+  i := 2;
+  while i < ParamCount do
+  begin
+    if not CheckParam(ParamStr(i), ParamStr(i+1)) then
+    begin
+      writeln('Invalid parameter: '+ParamStr(i));
```

Comment:
Would it be helpful to give the user feedback about `ParamsStr(i+1)`

Reply:
Well, I don't think so because that's a data parameter ... just knowing which parameter is unrecognised should be sufficient.